common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -184,6 +178,33 @@ protected SingleThreadEventExecutor(EventExecutorGroup parent, Executor executor
         rejectedExecutionHandler = ObjectUtil.checkNotNull(rejectedHandler, "rejectedHandler");
     }
 
+    /**
+     * Called from arbitrary non-{@link EventExecutor} threads prior to scheduled task submission.
+     * Returns {@code true} if the {@link EventExecutor} thread should be woken immediately to
+     * process the scheduled task (if not already awake).
+     * <p>
+     * If {@code false} is returned, {@link #afterFutureTaskScheduled(long)} will be called with
+     * the same value <i>after</i> the scheduled task is enqueued, providing another opportunity
+     * to wake the {@link EventExecutor} thread if required.
+     *
+     * @param deadlineNanos deadline of the to-be-scheduled task
+     *     relative to {@link AbstractScheduledEventExecutor#nanoTime()}
+     * @return {@code true} if the {@link EventExecutor} thread should be woken, {@code false} otherwise
+     */
+    protected boolean beforeFutureTaskScheduled(long deadlineNanos) {
```

Comment:
I wonder if we should maybe rename this to:

`beforeScheduledTaskSubmitted` just to match it better with the `schedule*` methods